build/hostlogswindows/installer/livenessprobe/livenessprobe.cpp

```diff
@@ -7,37 +7,127 @@
+    wchar_t* executableName = PathFindFileName(executableFullPath);
+
+    // Traverse through the list of processes in the snapshot
+    if (Process32First(processSnapshot, &processEntry))
     {
-        wprintf_s(L"INFO:File:%s exists indicates Config Map Updated since agent started.\n", argv[1]);
-        return FILESYSTEM_WATCHER_FILE_EXISTS;
+        do
+        {
+            if (!_wcsicmp(processEntry.szExeFile, executableName))
+            {
+                // For each process that matches our executableName, capture a snapshot of all modules
+                HANDLE moduleSnapshot = CreateToolhelp32Snapshot(TH32CS_SNAPMODULE, processEntry.th32ProcessID);
+
+                if (moduleSnapshot == INVALID_HANDLE_VALUE)
+                {
+                    throw std::runtime_error("Failed to create module snapshot"); 
+                }
+
+                MODULEENTRY32 moduleEntry{}; // struct to hold details of a module (exe) associated with a process
+                moduleEntry.dwSize = sizeof(MODULEENTRY32); // set size of struct
+
+                // If the module (exe) path matches with the given path, process is running
+                if (Module32First(moduleSnapshot, &moduleEntry) && !_wcsicmp(moduleEntry.szExePath, executableFullPath)) {
+                    CloseHandle(moduleSnapshot);
+                    CloseHandle(processSnapshot);
+                    return true;
+                }
+
+                CloseHandle(moduleSnapshot);
+            }
+        } while (Process32Next(processSnapshot, &processEntry));
     }
 
-    return SUCCESS;
+    CloseHandle(processSnapshot);
+    return false;
 }
+
+/**
+Usage: livesnessprobe.exe <monitoringAgentLauncherPath> [fileSystemWatcherTextFilePath]
+Description: 
+    - monitoringAgentLauncherPath: The mandatory path to the Monitoring Agent Launcher exe. The program will return exit code 1 if a process with this full path is not running.
+    - fileSystemWatcherTextFilePath: The optional path to the file system watcher text file. If provided, the program will return exit code 2 if this file exists.
+**/
+int wmain(int argc, wchar_t* argv[])
+{
+    try
+    {
+        if (argc < 2)
+        {
+            wprintf_s(L"ERROR: No arguments provided. "
+                    "Usage: livesnessprobe.exe <monitoringAgentLauncherPath> [fileSystemWatcherTextFilePath] "
+                    "Description: " 
+                    "- monitoringAgentLauncherPath: The mandatory path to the Monitoring Agent Launcher exe. The program will return exit code 1 if a process with this full path is not running. "
+                    "- watcherPath: The optional path to the watcher. If provided, the program will fail if this file exists.\n");
+            return UNEXPECTED_ERROR;
+        }
+        else if(argc == 2)
+        {
+            wprintf_s(L"INFO: Only the agentPath provided. The program won't check the watcher file.\n");
+        }
+        else if (argc == 3 && IsFileExists(argv[2]))
```

Comment:
Check the logic here.  If 3 args and configmap changes -> fail, else -> also fail. 
So the liveness probe always fails and only the startup probe checks if the MA is running

Reply:
Fixed